### PR TITLE
Allow asking the soft keyboard from outside the window widget

### PR DESCRIPTION
**Pull request by mshin-wolvic**
This patch fixes the issue that Chromium doesn't type nothing in the form. The problem was that the focused view notified from the system is not involved in the window widget which the keyboard widget is attached to. Since Chromium has a content view outside of the windows widget's layer, the focus view could be different with the window widget.

This patch allows asking the soft keyboard from the out side of the window widget.

### CODE REVIEW

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java [hunk 1]
```diff
@@ -1373,9 +1374,12 @@ public void restartInput(@NonNull WSession session, int reason) {
     }
 
     @Override
-    public void showSoftInput(@NonNull WSession session) {
+    public void showSoftInput(@NonNull WSession session, View requestView) {
+        mExternalFocusedView = requestView;
         if (mFocusedView != mAttachedWindow || getVisibility() != View.VISIBLE || mInputRestarted) {
-            post(() -> updateFocusedView(mAttachedWindow));
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java:1378]: Let's not change this as it's not needed and this is a more compact representation.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java [hunk 2]
```diff
@@ -1409,7 +1413,10 @@ public void run() {
 
     @Override
     public void onGlobalFocusChanged(View oldFocus, View newFocus) {
-        updateFocusedView(newFocus);
+        if (mExternalFocusedView == newFocus)
+            updateFocusedView(mAttachedWindow);
+        else
+            updateFocusedView(newFocus);
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java:1419]: We can simplify this to `updateFocusedView(mExternalFocusedView == newFocus ? mAttachedWindow : newFocus)`

**Review by svillar — CHANGES_REQUESTED** [review]
Excellent analysis. It's a pity that the hierarchy is the widget container is as it is, but I guess is the best we could do provided the big differences between chromium and gecko. At some point we should revisit that for the Chromium case.

I just have to minor nits and then we could land the patch.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java (hunk 1 above)
Comment (mshin-wolvic) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java:1378]: Done

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java (hunk 2 above)
Comment (mshin-wolvic) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/KeyboardWidget.java]: Done

**Review by svillar — APPROVED** [review]

**Pull request merged**